src/main/java/org/candlepin/subscriptions/jmx/OptInJmxBean.java

```diff
@@ -0,0 +1,81 @@
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.jmx;
+
+import org.candlepin.subscriptions.controller.OptInController;
+import org.candlepin.subscriptions.db.model.config.OptInType;
+import org.candlepin.subscriptions.utilization.api.model.OptInConfig;
+import org.candlepin.subscriptions.utilization.api.model.OptInConfigData;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.jmx.export.annotation.ManagedOperation;
+import org.springframework.jmx.export.annotation.ManagedOperationParameter;
+import org.springframework.jmx.export.annotation.ManagedResource;
+import org.springframework.stereotype.Component;
+
+/**
+ * Exposes the ability to perform OptIn operations.
+ */
+@Component
+@Profile({"api"})
+@ManagedResource
+public class OptInJmxBean {
+    private static final Logger log = LoggerFactory.getLogger(OptInJmxBean.class);
+
+    private final OptInController controller;
+
+    public OptInJmxBean(OptInController controller) {
+        this.controller = controller;
+    }
+
+    @ManagedOperation(description = "Delete opt in configuration")
+    @ManagedOperationParameter(name = "accountNumber", description = "Red Hat Account Number")
+    @ManagedOperationParameter(name = "orgId", description = "Red Hat Org ID")
+    public void deleteOptInConfig(String accountNumber, String orgId) {
+        controller.optOut(accountNumber, orgId);
+    }
+
+    @ManagedOperation(description = "Create an opt in configuration")
+    @ManagedOperationParameter(name = "accountNumber", description = "Red Hat Account Number")
+    @ManagedOperationParameter(name = "orgId", description = "Red Hat Org ID")
+    @ManagedOperationParameter(name = "enableTallySync", description = "Turn on Tally syncing")
+    @ManagedOperationParameter(name = "enableTallyReporting", description = "Turn on Tally reporting")
+    @ManagedOperationParameter(name = "enableConduitSync", description = "Turn on Conduit syncing")
+    public String createOptInConfig(String accountNumber, String orgId, boolean enableTallySync,
+        boolean enableTallyReporting, boolean enableConduitSync) {
+        log.debug("Creating OptInConfig over JMX for account {}, org {}", accountNumber, orgId);
+        OptInConfig config = controller.optIn(accountNumber, orgId, OptInType.JMX, enableTallySync,
+            enableTallyReporting, enableConduitSync);
+
+        OptInConfigData data = config.getData();
+
+        String text;
+        if (Boolean.TRUE.equals(data.getOptInComplete())) {
+            text = "Completed opt in for account %s and org %s";
+        }
+        else {
+            text = "Created opt in for account %s and org %s";
```

Comment:
This will never happen. optInComplete represents whether an AccountConfig/OrgConfig pair exists. Using the controller will never allow this to be false as it will create a new *Config if either doesn't exist.